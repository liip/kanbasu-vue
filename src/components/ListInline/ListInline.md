```vue
<list-inline size="large" divided>
  <li>One</li>
  <li>Two</li>
  <li>Three</li>
</list-inline>
```