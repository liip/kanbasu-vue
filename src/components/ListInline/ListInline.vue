<template>
  <ul :class="classNames">
    <slot></slot>
  </ul>
</template>

<script>
export default {
  name: 'ListInline',

  props: {
    /**
     * List items alignment:
     * `center, right, justified`
     */
    align: {
      type: String,
    },
    /**
     * List items gutter size:
     * `tight, tiny, small, large, huge`
     */
    size: {
      type: String,
    },
    /**
     * Add separators between items
     */
    divided: {
      type: Boolean,
    },
  },

  computed: {
    classNames() {
      return [
        'list-inline',
        { [`list-inline--${this.size}`]: !!this.size },
        { [`list-inline--${this.align}`]: !!this.align },
        { 'list-inline--divided': this.divided },
      ];
    },
  },
};
</script>
