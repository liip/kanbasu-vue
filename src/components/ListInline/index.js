import ListInline from './ListInline.vue';

export default ListInline;
