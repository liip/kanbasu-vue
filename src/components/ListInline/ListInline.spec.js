import { shallowMount } from '@vue/test-utils';
import ListInline from './ListInline.vue';

describe('ListInline.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(ListInline);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with size', () => {
    const wrapper = shallowMount(ListInline, {
      propsData: {
        size: 'small',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with alignment', () => {
    const wrapper = shallowMount(ListInline, {
      propsData: {
        align: 'center',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
  it('renders with dividers', () => {
    const wrapper = shallowMount(ListInline, {
      propsData: {
        divided: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
