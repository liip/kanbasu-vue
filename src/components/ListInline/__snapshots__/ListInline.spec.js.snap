// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListInline.vue renders 1`] = `<ul class="list-inline"></ul>`;

exports[`ListInline.vue renders with alignment 1`] = `<ul class="list-inline list-inline--center"></ul>`;

exports[`ListInline.vue renders with dividers 1`] = `<ul class="list-inline list-inline--divided"></ul>`;

exports[`ListInline.vue renders with size 1`] = `<ul class="list-inline list-inline--small"></ul>`;
